lib/sqlalchemy/cyextension/resultproxy.pyx

```diff
@@ -68,6 +73,14 @@ cdef class BaseRow:
         return self._data[index]
 
     cpdef _get_by_key_impl_mapping(self, key):
+        return self._real_get_by_key_impl_mapping(key)
+
+    cdef _real_get_by_key_impl_mapping(self, key):
```

Comment:
This probably makes sense but the rest wont't really save use anything since the dict lookup itself is expensive